docs/DeveloperGuide.md

```diff
@@ -262,30 +262,141 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `Medmoriser` and the **Actor** is the `user`, unless specified otherwise. Q&A stands for a question and answer pair)
 
-**Use case: Delete a person**
+**Use case: UC01 - Add a question and answer pair**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User adds a new Q&A into the database
+2. Medmoriser adds the new Q&A and displays a success message
+
+**Extensions**
+
+- 1a. The entered data is invalid
+
+  - 1a1. Medmoriser displays an error message
 
     Use case ends.
 
+
+
+**Use case: UC02 - View all Q&As**
+
+**MSS**
+
+1. User requests to view all Q&As
+2. Medmoriser shows a list of Q&As in the database
+
+
+
+**Use case: UC03 - Delete a Q&A**
+
+**MSS**
+
+1. User requests to view all Q&As
+
+2. Medmoriser shows a list of Q&As in the database
+
+3. User requests to delete a Q&A at a specific index on the list
+
+4. Medmoriser deletes the Q&A and displays success message 
+
+   Use case ends
+
 **Extensions**
 
-* 2a. The list is empty.
+- 2a. The list is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+- 3a. The given index is invalid.
+
+  - 3a1. Medmoriser shows an error message.
+
+    Use case resumes at step 2.
+
+**Use case: UC04 - Add a tag to a Q&A**
+
+**MSS**
+
+1. User requests to view all Q&As
+
+2. Medmoriser shows a list of Q&As in the database
+
+3. User requests to add a tag to a Q&A at a specific index on the list
+
+4. Medmoriser adds the specified tag to the Q&A and displays success message
+
+   Use case ends.
+
+**Extensions**
+
+- 2a. The list is empty.
+
+  Use case ends.
+
+- 3a. The given index is invalid.
+
+  - 3a1. Medmoriser shows an error message.
+
+    Use case resumes at step 2.
+
+- 3b. The given tag is invalid.
+
+  - 3b1. Medmoriser shows an error message.
+
+    Use case resumes at step 2.
+
+**Use case: UC05 - Edit a Q&A**
+
+**MSS**
+
+1. User requests to view all Q&As
+
+2. Medmoriser shows a list of Q&As in the database
+
+3. User requests to edit a Q&A at a specific index on the list
+
+4. Medmoriser edits the Q&A at the specified index according to the user's new input and displays success message
+
+   Use case ends
+
+**Extensions**
+
+- 2a. The list is empty.
+
+  Use case ends.
+
+- 3a. The given index is invalid.
+
+  - 3a1. Medmoriser shows an error message.
+
+    Use case resumes at step 2.
+
+- 3b. The edit request is invalid
+
+  - 3b1. Medmoriser shows an error message
+
+    Use case resumes at step 2.
+
+Use case: UC06 - Find a Q&A
```

Comment:
Might have missed out on bolding lines 383, 385, 393

Reply:
Ok I bolded the 3 lines